packages/SwingSet/src/types-internal.js

```diff
@@ -35,11 +48,48 @@ export {};
  * @property { boolean } enableSetup
  * @property { boolean } enablePipelining
  * @property { boolean } useTranscript
- * @property { number | 'never' } reapInterval
+ * @property { ReapDirtThreshold } reapDirtThreshold
  * @property { boolean } critical
  * @property { MeterID } [meterID] // property must be present, but can be undefined
  * @property { WorkerOptions } workerOptions
  * @property { boolean } enableDisavow
+ *
+ * @typedef ChangeVatOptions
+ * @property { number } [reapInterval]
+ */
+
+/**
+ * Reap/BringOutYourDead/BOYD Scheduling
+ *
+ * We trigger a BringOutYourDead delivery (which "reaps" all dead
+ * objects from the vat) after a certain threshold of "dirt" has
+ * accumulated. This type is used to define the thresholds for three
+ * counters: 'deliveries', 'gcKrefs', and 'computrons'. If a property
+ * is a number, we trigger BOYD when the counter for that property
+ * exceeds the threshold value. If a property is the string 'never' or
+ * missing we do not use that counter to trigger BOYD.
+ *
+ * Each vat has a .reapDirtThreshold in their vNN.options record,
+ * which overrides the kernel-wide settings in
+ * 'kernel.defaultReapDirtThreshold'
+ *
+ * @typedef { object } ReapDirtThreshold
+ * @property { number | 'never' } [deliveries]
+ * @property { number | 'never' } [gcKrefs]
+ * @property { number | 'never' } [computrons]
+ * @property { boolean } [never]
+ */
+
+/**
+ * Each counter in Dirt matches a threshold in
+ * ReapDirtThreshold. Missing values are treated as zero, so vats
+ * start with {} and accumulate dirt as deliveries are made, until a
+ * BOYD clears them.
+ *
+ * @typedef { object } Dirt
+ * @property { number } [deliveries]
+ * @property { number } [gcKrefs]
+ * @property { number } [computrons]
```

Comment:
"gcKrefs" reads to me like a verb. Suggestion:
```suggestion
 * @typedef { object } Dirt
 * @property { number } [deliveries]
 * @property { number } [downgradedKrefs]
 * @property { number } [computrons]
```
(or similar)

Or perhaps better, identify all the properties as counts:
```suggestion
 * @typedef { object } Dirt
 * @property { number } [deliveryCount]
 * @property { number } [krefCount]
 * @property { number } [computronCount]
```

Reply:
Hm, that *does* read better, although a bit more verbose. I do need to maintain the "GC" -ness of the count, though: a plain `krefCount` could mean how many krefs we send into a vat, which is an entirely different (and completely plausible) thing that we might want to trigger a BOYD.

I'll try `deliveryCount` / `gcKrefCount` / `computronCount` and see how it reads.